dlt/pipeline/pipeline.py

```diff
@@ -856,29 +856,25 @@ def _extract_source(self, storage: ExtractorStorage, source: DltSource, max_para
         source_schema = source.schema
         source_schema.update_normalizers()
 
-        extract_id = extract_with_schema(storage, source, source_schema, self.collector, max_parallel_items, workers)
-
         # if source schema does not exist in the pipeline
         if source_schema.name not in self._schema_storage:
-            # save schema into the pipeline
-            self._schema_storage.save_schema(source_schema)
+            # create a new schema with the right name
+            self._schema_storage.save_schema(Schema(source_schema.name))
+
+        # get the schema from pipeline storage and merge stuff from the new schema
+        pipeline_schema = self._schema_storage[source_schema.name]
+        pipeline_schema.update_schema(source_schema)
```

Comment:
probably makes sense to update settings and configs but not tables. that will happen later in extract and will be controlled by contracts

Reply:
There are tables already on the source_schema when it comes from the source before we run extract in some cases, so we need to merge these in too.

Reply:
OK - with this implementation pipeline_schema is a reference for the contract. so if there are changes done in the code (ie. schema is modified directly - they will be rejected if contract says so. hmmm I think it is OK.